source/Octopus.Server.Client/Repositories/Async/VariableSetRepository.cs

```diff
@@ -1,16 +1,31 @@
 using System;
 using System.Collections.Generic;
+using System.Threading;
 using System.Threading.Tasks;
 using Octopus.Client.Model;
 
 namespace Octopus.Client.Repositories.Async
 {
     public interface IVariableSetRepository : IGet<VariableSetResource>, IModify<VariableSetResource>, IGetAll<VariableSetResource>
     {
+        [Obsolete("Please use the overload with cancellation token instead.", false)]
         Task<string[]> GetVariableNames(string projects, string[] environments, string gitRef = null);
+        Task<string[]> GetVariableNames(string projects, string[] environments, CancellationToken cancellationToken);
+        Task<string[]> GetVariableNames(string projects, string[] environments, string gitRef, CancellationToken cancellationToken);
+        
+        [Obsolete("Please use the overload with cancellation token instead.", false)]
         Task<VariableSetResource> GetVariablePreview(string project, string channel, string tenant, string runbook, string action, string environment, string machine, string role);
+        Task<VariableSetResource> GetVariablePreview(string project, string channel, string tenant, string runbook, string action, string environment, string machine, string role, CancellationToken cancellationToken);
+        
+        [Obsolete("Please use the overload with cancellation token instead.", false)]
         Task<VariableSetResource> GetVariablePreview(ProjectResource projectResource, string channel = null, string tenant = null, string runbook = null, string action = null, string environment = null, string machine = null, string role = null, string gitRef = null);
+        Task<VariableSetResource> GetVariablePreview(ProjectResource projectResource, CancellationToken cancellationToken);
+        Task<VariableSetResource> GetVariablePreview(ProjectResource projectResource, string channel, string tenant, string runbook, string action, string environment, string machine, string role, string gitRef, CancellationToken cancellationToken);
```

Comment:
Favouring tightening of contract, instead of **polluting interface** with overloads to support existing optionals